risc0/zkp/rust/src/core/ntt.rs

```diff
@@ -77,31 +73,35 @@ pub fn bit_reverse<T: Copy>(io: &mut [T]) {
 }
 
 #[inline]
-fn fwd_butterfly_0<T>(_: &mut [T], _: usize) {
+fn fwd_butterfly_0<B, T>(_: &mut [T], _: usize) {
     // no-op base case
 }
 
 #[inline]
-fn rev_butterfly_0<T>(_: &mut [T]) {
+fn rev_butterfly_0<B, T>(_: &mut [T]) {
     // no-op base case
 }
 
+// TODO: This generates butterfly functions up to $n = 32, but will panic if $n
```

Comment:
I am generating butterfly functions up to `$n = 32`, but for some fields (e.g. baby bear) these aren't all legal and some will panic. Would it be better to return `Option`s and handle bad calls?